docs: update API versions, kinds, and quickstart next steps (#141)
openmcp-project/docs: docs/operators/01-quickstart.md

ps-managedcontrolplane-init-t46cn         0/1     Completed   0          113s
sp-crossplane-84f8bbfb58-24lwl            1/1     Running     0          79s
sp-crossplane-init-z657s                  0/1     Completed   0          113s
sp-flux-844b667b9c-mhlqn                  1/1     Running     0          73s
sp-flux-init-x5pxw                        0/1     Completed   0          112s
sp-kro-ffbdcf787-zjrfz                    1/1     Running     0          83s
sp-kro-init-rf5vk                         0/1     Completed   0          113s
sp-ocm-756946b9dd-g67sd                   1/1     Running     0          70s
sp-ocm-init-6cp5c                         0/1     Completed   0          111s
```

:::

In this output we can see that openmcp-operator and multiple other services like cluster-provider-kind (cp-kind) and service providers such as Crossplane, Flux, Kro and OCM are running.

:::note Error
We might see that Platform Service Gateway (`ps-gateway-5db88d9474-6sxsp`) has an ERROR.

```shell
...
Error: applying resources: applying Unstructured//gateway: no matches for kind "GatewayServiceConfig" in version "gateway.openmcp.cloud/v1alpha1"
```

> Please wait for a couple of seconds and then try execute `ocpctl env apply local` again. This will restart the Pod. All Pods above should be running eventually.

Reviewer: long-term we should make `ocpctl` stable enough so errors like this dont have to be documented.

Author: 💯 I am totally on your side.

:::

### Configure allowed Flux versions

> In order that end users can request Flux getting installed in a `ControlPlane`, as Platform Owner, we need to make sure to configure allowed versions of Flux. We can configure them via a `ProviderConfig`:

Reviewer 2: nit: slight rephrasing suggestion to make it a little easier to read:
```suggestion
To enable end users to request Flux for a `ControlPlane`, as Platform Owner, we need to make sure to configure allowed versions of Flux. We can configure them via a `ProviderConfig`:
```

Feel free to ignore this.

Author: I like your idea. Addressed with [`8ffbd62` (this PR)](https://github.com/openmcp-project/docs/pull/141/commits/8ffbd622da5a37213bb35bb821fc76c5de51ef9a)

You might receive the following error message:

```shell
error: resource mapping not found for name: "flux" namespace: "" from "STDIN": no matches for kind "ProviderConfig" in version "flux.services.open-control-plane.io/v1alpha1"
ensure CRDs are installed first
```
Please wait for a couple of seconds and then try again. Continue when the output says: `providerconfig.flux.services.open-control-plane.io/flux created`.

> :::

Reviewer: This error should never appear, because the `ocpctl env apply` will quit with an *error* or with *success*. And in success-case the resources will all be applied.

Author: Good catch! You are right. This error note can be completely removed. Just tested it locally again and there is no way I run into this error ✅ Removed with [`122a7a2` (this PR)](https://github.com/openmcp-project/docs/pull/141/commits/122a7a223193b0dd057de7ab28cf7b0973b17d2b)

```

:::

But we can also apply a new `ServiceProvider` to our platform to offer e.g. External Secrets Operator to end users:

:::apply-to-platform

```shell
kubectl config use-context kind-local-platform
kubectl apply -f - <<EOF
apiVersion: openmcp.cloud/v1alpha1
kind: ServiceProvider
metadata:
  name: externalsecretsoperator
  namespace: openmcp-system
spec:
  image: ghcr.io/openmcp-project/images/service-provider-external-secrets:v1.0.0
EOF
```

We can look up the status of the installation by executing:
```shell
kubectl config use-context kind-local-platform
kubectl get serviceproviders
```

The output looks like this:

```shell
NAME                      PHASE
crossplane                Ready
externalsecretsoperator   Progressing
flux                      Ready
kro                       Ready
ocm                       Ready
```

:::

Next, we need to configure these `ServiceProviders` via their `ProviderConfig` API to rule which versions end users can install.

:::note
MORE COMING SOON
:::

### Managed team access
Learn how [Projects and Workspaces](/users/concepts/projects-and-workspaces) let you organize teams and `ControlPlanes`.

:::note
MORE COMING SOON
:::

> ### Deploy on real infrastrcture

Reviewer 2: ```suggestion
### Deploy on real infrastructure
```

Author: Good catch! Fixed with [`c939d67` (this PR)](https://github.com/openmcp-project/docs/pull/141/commits/c939d67b5b6e28e5f8706b27ce96e38a50e1af38)
